sympy/printing/latex.py

```diff
@@ -1415,6 +1417,55 @@ def _print_ZeroMatrix(self, Z):
     def _print_Identity(self, I):
         return r"\mathbb{I}"
 
+    def _print_NDimArray(self, expr):
+
+        mat_str = self._settings['mat_str']
+        if mat_str is None:
+            if self._settings['mode'] == 'inline':
+                mat_str = 'smallmatrix'
+            else:
+                if (expr.rank() == 0) or ((expr.shape[-1] <= 10) == True):
```

Comment:
Would it be possible to produce a sensible latex string if `shape[-1]` were symbolic?